entries-api/support.js

```diff
@@ -89,3 +87,43 @@ const FILE_PATHS = [
   '/upload//file.txt',
   'subdir/./../file.txt',
 ];
+
+// ----------------------------------------
+// Helpers
+// ----------------------------------------
+
+// Wrapper for FileSystemDirectoryReader that yields all entries via a
+// Promise.
+
+function getEntriesAsPromise(dirEntry) {
+  return new Promise((resolve, reject) => {
+    let result = [];
+    let reader = dirEntry.createReader();
+    let doBatch = () => {
+      reader.readEntries(entries => {
+        if (entries.length > 0) {
+          entries.forEach(e => result.push(e));
+          doBatch();
+        } else {
+          resolve(result);
+        }
+      }, reject);
+    };
+    doBatch();
+  });
+}
+
+
+// Wrapper for FileSystemDirectoryReader that yields a single entry by
+// name via a callback. Can be used instead of getFile() or
+// getDirectory() since not all implementations support those.
+
+function getChildEntry(dirEntry, name, callback) {
+  getEntriesAsPromise(dirEntry)
+    .then(function(entries) {
```

Comment:
nit: entries => {} might be more consistent with the rest of these tests?

Reply:
Thanks, meant to fix this and forgot. Done!